File: source_formatting/commentary_parentheses.py
#!/usr/bin/python
# -*- coding: utf-8 -*-

from source_formatting.html_parser import BaseHtmlTranslator

class CommentaryParenthesesTransformer(BaseHtmlTranslator):
    english_name_ignore = ("Steinsaltz", "Shulchan Arukh", "Mishneh Torah")

    def __init__(self):
        super().__init__()

    def handle_starttag(self, tag, attrs):
        self.append_start_tag(tag, attrs)

    def handle_endtag(self, tag):
        self.append_end_tag(tag)

    def handle_data(self, data):
        self._out.append(
            data.replace('(', '<span class="parenthesized">(').replace(')', ')</span>'))

File: source_formatting/commentary_parentheses.ts
import {Attributes, HtmlVisitor} from "./html_visitor";

export class CommentaryParenthesesTransformer extends HtmlVisitor {
  englishNameIgnore = new Set(["Steinsaltz", "Shulchan Arukh", "Mishneh Torah"])

  shouldRun(input: string): boolean {
    // TODO: this isn't perfect since it doesn't checked for balancedness
    return input.includes("(") || input.includes(")");
  }

  visitStartTag(tag: string, attributes: Attributes): void {
    this.appendStartTag(tag, attributes);
  }

  visitEndTag(tag: string): void {
    this.appendEndTag(tag);
  }

  visitText(text: string): void {
    this.appendText(
      text
        .replace(/\(/g, '<span class="parenthesized">(')
        .replace(/\)/g, ')</span>'));
  }
}
